tests/unit/amo/components/TestHeader.js

```diff
@@ -77,17 +76,13 @@ describe(__filename, () => {
 
   it('allows a signed-in user to log out', () => {
     const { store } = dispatchSignInActions({ username: 'babar' });
-    const wrapper = renderHeader({ store });
-    const mockApi = sinon.mock(api);
+    const handleLogOut = sinon.stub();
 
-    mockApi
-      .expects('logOutFromServer')
-      .once()
-      .returns(Promise.resolve());
+    const wrapper = renderHeader({ store, handleLogOut });
 
     const onClick = wrapper.find(DropdownMenuItem).last().prop('onClick');
     onClick(createFakeEvent());
 
-    mockApi.verify();
+    sinon.assert.called(handleLogOut);
```

Comment:
This needs to also make sure `handleLogOut` was called with `{ api }` since that's a crucial part of its signature. This is the type of thing Flow can be really good for. It would tell you if you're calling `handleLogOut()` incorrectly even though it's a stub.

Reply:
ah sure!